hostports/watcher.go

```diff
@@ -172,61 +193,73 @@ func (w *watcher) onChange(version string) error {
 		return err
 	}
 
-	for _, container := range containers {
-		network := networks[container.NetworkUUID]
-		bridge := ""
+	services, err := w.c.GetServices()
+	if err != nil {
+		return err
+	}
+
+	mangedNetworks, localRouters := getLocalManagedInfo(networks, host, services)
 
+	// For iptables filter table rules
+	for _, n := range mangedNetworks {
+		if c, ok := localRouters[n.UUID]; !ok || len(c.Ports) == 0 { // For bridged and routed
+			bridge, bridgeSubnet := getBridgeInfo(n, host)
+			newChainRules[n.UUID] = ChainRule{
```

Comment:
This is really confusing, using the network UUID as the key. Needs some refactoring here. Probably add one more argument to apply and pass a different map of rules, but let's not overload the same map and try to differentiate based on the presence of protocol being null or not.